### PR TITLE
refactor(oob)!: make label optional

### LINKED ISSUE #1524: Make `label` optional in `OutOfBandInvitation`
It is not required according to the specification.

Also need to update the `parseInvitation` method to not add an empty string as the label when parsing legacy connectionless invitations (which don't have labels).

### PR DESCRIPTION
**Pull request by TimoGlastra**
Fixes #1524

### CODE REVIEW

File: packages/core/src/modules/oob/helpers.ts [hunk 1]
```diff
@@ -45,15 +45,17 @@ export function convertToOldInvitation(newInvitation: OutOfBandInvitation) {
   if (typeof service === 'string') {
     options = {
       id: newInvitation.id,
-      label: newInvitation.label,
+      // label is optional
+      label: newInvitation.label ?? '',
```
Comment (berendsliedrecht) [inline packages/core/src/modules/oob/helpers.ts:49]: If it is optional, why do we still fallback to an empty string?

**Review by berendsliedrecht — APPROVED** [review]

File: packages/core/src/modules/oob/helpers.ts (hunk 1 above)
Comment (TimoGlastra) [reply packages/core/src/modules/oob/helpers.ts:49]: Because it's not optional in an conection protocol invitation, only the oob invitation.

**Comment by genaris** [comment]
@TimoGlastra what about merging this before #1705 so we can create an OOB Invitation without label?

**Comment by TimoGlastra** [comment]
Sounds good, it's ready to merge if CI passes :)

**Pull request merged**